doc/source/dev/toolchain.rst

```diff
@@ -0,0 +1,136 @@
+3.3, 3.4                10.0           C90 + some of C99
+2.7, 3.0, 3.1, 3.2       9.0           C90
+===================   ==============   ===================
+
+
+C Language Standards
+--------------------
+[*NOTE: I'm unaware of any official SciPy decisions on the support of C standards.
+This table shows what it could look like.*]
+
+================  =========================================
+ Date              C Standard
+================  =========================================
+ <= 2018           C90
+ 2019              C90 for old code, may allow C99 for new
+ 2020              C99
+ No decision       C11
+ No decision       C17, C18
+================  =========================================
+
+Some C99 features would be useful for scientific programming, in particular better support of
+IEEE 754 [2]_.  Experience has shown that not all features are supported equally well across
+all platforms, SciPy has a small include file ``scipy/_lib/_c99compat.h`` which provides
+access to a few functions.  Use in conjunction with ``<numpy/npy_math.h>``.
+
+================================  ========================================================
+ Feature                           Workaround
+================================  ========================================================
+ isnan(), isinf(), isfinite()      Use sc_isnan(), sc_isinf(), sc_isfinite()
+ NAN                               Use NPY_NAN (*almost* equivalent)
+ inline functions                  Make static functions and place in an include .h file
+ mid-block variable declarations   Declare variables at the top of block
+================================  ========================================================
+
+Fortran Compilers
+-----------------
+
+[*What is the requirement on a Fortran compiler*]
+
+Cython Compiler
+---------------
+
+SciPy always requires a recent Cython compiler. As of v1.12.0, the minumum version is 0.28.5? 0.29?
+
+Numpy
+-----
+SciPy depends on NumPy but releases of SciPy are not tied to releases of NumPy.
+SciPy attempts to be compatible with the 4 previous releases of NumPy.
+In particular, SciPy can not rely on features of just the latest NumPy, but needs to be
+written using what is common in all of those 4 releases.
+
+========  ========================    ===========================
+ Python     minimum Numpy version     maximum Numpy version
+========  ========================    ===========================
+3.7         1.13.1                     >= 1.16.x
+3.6         1.12.1                     >= 1.16.x
+3.5         1.9.3                      >= 1.16.x
+2.7         1.8.2                      >= 1.16.x
+========  ========================    ===========================
+
+
+
+Other Libraries
+---------------
+
+- LAPACK: >= 3.4.1
+- OpenBLAS: A recent version
+
+
+Building the Documentation
+--------------------------
+
+- Sphinx: whatever recent versions work. >= 1.6.6.
+- numpydoc: whatever recent versions work. >=  0.8.0.
+- matplotlib: ?
```

Comment:
good question. was never important enough to define a minimum version, since we use so few features of it. at least >=2.0 I'd say, although a number of 1.x releases will likely work without any problems

Reply:
I put in the 2.0 matplotlib min version guidance Ralf suggested